docs/tutorials/java/audiobook/index.md

```diff
@@ -0,0 +1,982 @@
+        } catch (IOException | IllegalArgumentException | SecurityException e) {
+            fail("Unable to create temporary work file", "FILE_ERROR");
+        }
+        return null;
+    }
+
+    byte[] textToSpeech(String text) throws IOException {
+        String apiEndpoint = "https://api.openai.com/v1/audio/speech";
+
+        OkHttpClient client = new OkHttpClient();
+
+        JSONObject json = new JSONObject();
+        json.put("model", "tts-1");
+        json.put("input", text);
+        json.put("voice", "nova"); // see https://platform.openai.com/docs/guides/text-to-speech/voice-options
+        json.put("response_format", "mp3");
+
+        MediaType mediaType = MediaType.get("application/json; charset=utf-8");
+        RequestBody body = RequestBody.create(json.toString(), mediaType);
+
+        Request request = new Request.Builder()
+            .url(apiEndpoint)
+            .post(body)
+            .addHeader("Authorization", "Bearer " + bearerToken)
+            .build();
+
+        try (Response response = client.newCall(request).execute()) {
+            if (!response.isSuccessful()) {
+                throw new IOException("Unexpected code " + response);
+            }
+            return response.body().bytes();
+        }
+    }
+
+    @Override
+    public void process(String chunk, Path outputPath) {
+        byte[] audio;
+
+        try {
+            audio = textToSpeech(chunk);
+        } catch (IOException e) {
+            throw new RuntimeException(e);
+        }
+
+        try {
+            Files.write(outputPath, audio,
+                            java.nio.file.StandardOpenOption.CREATE,
+                            java.nio.file.StandardOpenOption.APPEND);
+        } catch (IOException e) {
+            throw fail("Unable to write to output file", "FILE_ERROR");
+        }
+    }
+
+    @Override
+    public String moveOutputFileToPlace(Path tempPath) {
+        Path newPath = null;
+        String extension = ".mp3";
+        try {
+            String baseName = FilenameUtils.getBaseName(canonicalPath.toString());
+            Path parentDir = canonicalPath.getParent();
+            newPath = parentDir.resolve(Paths.get(baseName + extension));
+            int suffixCounter = 1;
+            while (Files.exists(newPath)) {
+                String newFileName = baseName + "-" + suffixCounter + extension;
+                newPath = parentDir.resolve(newFileName);
+                suffixCounter += 1;
+            }
+            Files.move(tempPath, newPath, StandardCopyOption.REPLACE_EXISTING);
+        } catch (InvalidPathException | IOException e) {
+            throw fail("Unable to move output file to destination", "FILE_ERROR");
+        }
+        return newPath.toString();
+    }
+}
+```
+<!--SNIPEND-->
+
+  </TabItem>
+</Tabs>
+
+</details>
+
+Activities, like the `TTSActivities` class, handle potentially unreliable parts of your code, such as calling APIs or working with file systems.
```

Comment:
Could add a link to activities in the encyclopedia.

Reply:
As I should. Adding!